services/worker-manager/src/api.js

```diff
@@ -5,6 +5,8 @@ const { ApiError, Provider } = require('./providers/provider');
 const { UNIQUE_VIOLATION } = require('taskcluster-lib-postgres');
 const { WorkerPool, WorkerPoolError, Worker } = require('./data');
 const { createCredentials } = require('./util');
+const { joinTaskQueueId, addSplitFields } = require('../../queue/src/utils');
+const { TaskQueue } = require('../../queue/src/data');
```

Comment:
This is where it gets interesting :) 
As soon as we start to introduce such dependencies, we will have one double-micro-service :D 
If we really need this class, we can copy it like you did with the schema. Having copy-pasted class would be better than having highly coupled services imho.